module/zstd/zstd.c

```diff
@@ -181,6 +187,12 @@ static int pool_count = 16;
 static struct zstd_fallback_mem zstd_dctx_fallback;
 static struct zstd_pool *zstd_mempool_cctx;
 static struct zstd_pool *zstd_mempool_dctx;
+static int __init zstd_meminit(void);
+
+#ifdef DEBUG
+int mem_init_done = 0;
+uint8_t test_level = 0;
```

Comment:
very bad idea. the code is threaded. static members have bad consequences. its not thread safe. test_level is updated on each compression / decompression call and the level can be different for each call. but other threads depend still on a different value

Reply:
yup, I'll drop the testing stuff anyways now..

Reply:
oh, wait.... just to clarify that... no it's not a problem here, since the test level is exactly the same for one whole test run ;) but in theory it's wrong, yes

Reply:
@c0d3z3r0 not at all. assume you are running 2 tests the second test uses zstd but a different level. but the first test has still delayed writes running. if a test is finished it doesnt mean that all data is already written

Reply:
hmmm, indeed... ack